src/main/java/com/louis/HashTable.java

```diff
@@ -0,0 +1,219 @@
+package com.louis;
+
+import com.google.common.base.Preconditions;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.List;
+import java.util.ArrayList;
+import java.lang.reflect.Array;
+
+/**
+ * Implements a hashtable.
+ *
+ * Hashtables implement the associative array abstract data type.
+ * The tables contain keys and associated values.
+ * A hash function computes the location of a value for a given key.
+ *
+ * @author Louis Ashton (louisashton@live.com)
+ */
+public class HashTable<K, V> {
+
+    private static final int MIN_CAPACITY = 32;
+    private static final double RESIZE_LOAD = 0.5;
+    private static final double MAX_LOAD = 0.75;
+    private static final double MIN_LOAD = 0.25;
+
+    private HashEntry[] table;
+    private int currentSize;
+    private int currentCapacity;
+
+
+    /**
+     * Creates a new HashTable.
+     */
+    @SuppressWarnings("unchecked")
+    public HashTable() {
+        currentSize = 0;
+        currentCapacity = MIN_CAPACITY;
+        table = (HashEntry[]) Array.newInstance(HashEntry.class, currentCapacity);
+    }
+
+    /**
+     * Defines a HashTable entry.
+     * Similar to java.util.AbstractMap.SimpleEntry.
+     */
+    private class HashEntry {
+
+        private final K key;
+        private final V value;
+
+        /**
+         * Creates a new HashEntry.
+         */
+        HashEntry(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+
+        /**
+         * Returns the key.
+         * @return Returns the entries' key.
+         */
+        public K getKey() {
+            return key;
+        }
+
+        /**
+         * Returns the value.
+         * @return Returns the entries' value.
+         */
+        public V getValue() {
+            return value;
+        }
+    }
+
+    /**
+     * Returns the value associated with a key.
+     *
+     * @param key is the specified key.
+     * @return Returns the value associated with the key; null if no such value.
```

Comment:
`@return The value associated with the key or null if no value was stored for that key.